wkcuber/api/MagDataset.py

```diff
@@ -0,0 +1,135 @@
+from os.path import join
+
+from wkw import wkw
+
+import wkcuber.api as api
+from wkcuber.api.Slice import WKSlice, TiffSlice
+from wkcuber.api.TiffData.TiffMag import TiffMagHeader
+
+
+class MagDataset:
+    def __init__(self, layer, name):
+        self.mag = {}
+        self.layer = layer
+        self.name = name
+        self.header = self.get_header()
+
+        file_path = join(self.layer.dataset.path, self.layer.name, self.name)
+        size = self.layer.dataset.properties.data_layers[
+            self.layer.name
+        ].get_bounding_box_size()
+        offset = self.layer.dataset.properties.data_layers[
+            self.layer.name
+        ].get_bounding_box_offset()
+        self.slice = self.__get_slice_type__()(
```

Comment:
1) Why can't you use `get_slice()` here? If that method accesses `self.header` then the signature of `get_slice` should match your requirements here.
2) Maybe it's a good idea to add an `is_bounded` parameter to `get_slice`? That way, you don't need to mutate the slice object afterwards.